tests/system/test_system.py

```diff
@@ -890,6 +891,58 @@ def test_collection_group_queries_filters(client, cleanup):
     assert found == set(["cg-doc2"])
 
 
+def test_partition_query_no_partitions(client, cleanup):
+    collection_group = "b" + UNIQUE_RESOURCE_ID
+
+    # less than minimum partition size
+    doc_paths = [
+        "abc/123/" + collection_group + "/cg-doc1",
+        "abc/123/" + collection_group + "/cg-doc2",
+        collection_group + "/cg-doc3",
+        collection_group + "/cg-doc4",
+        "def/456/" + collection_group + "/cg-doc5",
+    ]
+
+    batch = client.batch()
+    for doc_path in doc_paths:
+        doc_ref = client.document(doc_path)
+        batch.set(doc_ref, {"x": 1})
+        cleanup(doc_ref.delete)
+
+    batch.commit()
+
+    query = client.collection_group(collection_group)
+    streams = [partition.stream() for partition in query.get_partitions(3)]
+    snapshots = itertools.chain(*streams)
+    found = [snapshot.id for snapshot in snapshots]
+    expected = ["cg-doc1", "cg-doc2", "cg-doc3", "cg-doc4", "cg-doc5"]
+    assert found == expected
+
+
+def test_partition_query(client, cleanup):
+    collection_group = "b" + UNIQUE_RESOURCE_ID
+    n_docs = 128 * 2 + 127  # Minimum partition size is 128
+    parents = itertools.cycle(("", "abc/123/", "def/456/", "ghi/789/"))
+    batch = client.batch()
+    cleanup_batch = client.batch()
+    for i, parent in zip(range(n_docs), parents):
+        doc_path = parent + collection_group + f"/cg-doc{i:03d}"
+        doc_ref = client.document(doc_path)
+        batch.set(doc_ref, {"x": i})
+        cleanup_batch.delete(doc_ref)
+
+    cleanup(cleanup_batch.commit)
+    batch.commit()
+
+    query = client.collection_group(collection_group)
+    streams = [partition.stream() for partition in query.get_partitions(3)]
+    snapshots = itertools.chain(*streams)
+    found = [snapshot.id for snapshot in snapshots]
+    found.sort()
```

Comment:
Order that documents are returned in is odd and seems to be related to underlying implementation details.

For example, if docs are number sequentially starting at 0 and we get back 3 cursors (4 partitions), the first partition will be:

1, 5, 9, 13, ..., 381, 0, 4, 8, 12

I assume that's expected and we shouldn't monkey with it.